src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -714,21 +660,31 @@ uct_cuda_copy_md_query_attributes(uct_cuda_copy_md_t *md, const void *address,
         return UCS_ERR_NO_DEVICE;
     }
 
-    uct_cuda_copy_sync_memops(md, address, is_vmm);
+    if (is_vmm) {
```

Comment:
We will retain primary cuda device context even if the device is `CU_DEVICE_CPU`. It is better to retain it right before the usage.